### PR TITLE
feat(schema): add resource and scope table

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Implement resource-servers and resource-scopes table along with some basic queries

please check the notion [doc](https://www.notion.so/silverhand/Resource-Indicators-6dfc5634e9fb496e9219de09c25494a4) for detail context

## Testing

Run the create table sql on local postgresql server:

resource_services table:
image

resource_scopes table:
image

image

### CODE REVIEW

File: packages/core/src/queries/resources.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import { buildInsertInto } from '@/database/insert-into';
+import pool from '@/database/pool';
+import { buildUpdateWhere } from '@/database/update-where';
+import { convertToIdentifiers, OmitAutoSetFields } from '@/database/utils';
+import RequestError from '@/errors/RequestError';
+
+const { table, fields } = convertToIdentifiers(ResourceServers);
+
+export const findAllResources = async () =>
```
Comment (xiaoyijun) [inline packages/core/src/queries/resources.ts:12]: `ResourceServers` would be better?
`resource` is tend to represent a specific entity of data.
something like "resource servers provide resources for user`

**Review by xiaoyijun — CHANGES_REQUESTED** [review]

File: packages/schemas/tables/resource_scopes.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+create table resource_scopes (
+    id varchar(24) not null,
+    name varchar(64) not null,
+    description text not null,
+    resource_id varchar(24) not null,
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/schemas/tables/resource_scopes.sql:5]: Did some cross checking. Intend to keep the `resource_id` naming convention here, instead update the `resource-servers` name using just `resources` as for most of the cases we use general resource to indicate the data. i.e. in URIParam it is `resource` and the indicators are called `resourceIndicator` & use the keywords `resource` cross all OAuth context & oidc-provider.

File: packages/core/src/queries/resources.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/queries/resources.ts:12]: ditto

File: packages/core/src/queries/scopes.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { sql } from 'slonik';
+
+import { buildInsertInto } from '@/database/insert-into';
+import pool from '@/database/pool';
+import { convertToIdentifiers } from '@/database/utils';
+import RequestError from '@/errors/RequestError';
+
+const { table, fields } = convertToIdentifiers(ResourceScopes);
+
+export const findAllScopesWithResourceId = async (resourceId: string) =>
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/queries/scopes.ts:11]: ditto

File: packages/schemas/tables/resource_scopes.sql [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    primary key (id),
+    constraint fk_resource
+        foreign key (resource_id)
+            references resource_servers(id)
+);
+
+create unique index resource_scopes__resource_id_name
+on resource_scopes (
+    name,
+    resource_id
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/schemas/tables/resource_scopes.sql:15]: updated

File: packages/schemas/tables/resource_scopes.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/schemas/tables/resource_scopes.sql:5]: 👌

**Comment by xiaoyijun** [comment]
Overall LGTM

**Review by wangsijie — APPROVED** [review]
LGTM

**Pull request merged**